include/cantera/numerics/funcs.h

```diff
@@ -28,6 +29,54 @@ namespace Cantera
  */
 doublereal linearInterp(doublereal x, const vector_fp& xpts,
                         const vector_fp& fpts);
-}
 
+//! Numerical integration of a function using the trapezoidal rule.
+/*!
+ * Vector x contanins a monotonic sequence of grid points, and
+ * Vector f contains function values defined at these points.
+ * The size of x and f must be the same.
+ *
+ * @param  f vector of function value
+ * @param  x vector of function coordinate
+ */
+double trapezoidal(const Eigen::ArrayXd& f, const Eigen::ArrayXd& x);
+
+//! Numerical integration of a function using the simpson's rule.
+//! Only for odd number of points.
+/*!
+ * Vector x contanins a monotonic sequence of grid points, and
+ * Vector f contains function values defined at these points.
+ * The size of x and f must be the same.
+ *
+ * @param  f vector of function value
+ * @param  x vector of function coordinate
+ */
+double basicSimpson(const Eigen::ArrayXd& f, const Eigen::ArrayXd& x);
```

Comment:
I think this can just be a private function defined only in the `.cpp` file -- there's no reason someone should call this directly instead of the general version below.

Reply:
Where should I put the docstring?

Reply:
You can put some documentation with the implementation itself. I don't think it's critical for a local method like this to have a proper Doxygen docstring, though, and it's probably best if it doesn't actually show up in the HTML documentation.